pyrobosim/pyrobosim/planning/pddlstream/primitives.py

```diff
@@ -108,6 +108,54 @@ def sample_motion(planner, p1, p2):
         yield (path,)
 
 
+def sample_grasp_pose(grasp_gen, obj, p_obj, p_robot,
+                      front_grasps=True, top_grasps=True, side_grasps=False):
+    """
+    Samples feasible grasps for an object given its pose and the relative robot pose.
+
+    :param grasp_gen: Grasp generator object
+    :type grasp_gen: Planner
+    :param obj: Target object
+    :type obj: :class:`pyrobosim.core.objects.Object`
+    :param p_obj: Object pose.
+    :type p_obj: :class:`pyrobosim.utils.pose.Pose`
+    :param p_robot: Robot pose.
+    :type p_robot: :class:`pyrobosim.utils.pose.Pose`
+    :param front_grasps: Enable front grasps
+    :type front_grasps: bool, optional
+    :param top_grasps: Enable top grasps
+    :type top_grasps: bool, optional
+    :param side_grasps: Enable side grasps
+    :type side_grasps: bool, optional
+    :return: Generator yielding tuple containing a grasp
+    :rtype: generator[tuple[:class:`pyrobosim.manipulation.grasping.Grasp`]]
+        
+    """
+    # Get the object cuboid pose assuming the object is at pose p_obj
+    cuboid_pose = Pose.from_transform(
+        np.matmul(obj.cuboid_pose.get_transform_matrix(),
+                  p_obj.get_transform_matrix(),      
+        )
+    )
+
+    # Generate grasps up front and yield them as requested
+    grasps = grasp_gen.generate(
+        obj.cuboid_dims, cuboid_pose, p_robot,
+        front_grasps=front_grasps,
+        top_grasps=top_grasps,
+        side_grasps=side_grasps
+    )
+    for g in grasps:
+        # Convert the grasp origin from object to world cooridnates
+        g.origin = Pose.from_transform(
+            np.matmul(
+                g.origin.get_transform_matrix(),
+                cuboid_pose.get_transform_matrix(),
+            )
+        )
```

Comment:
Notice how grasps are by default expressed in object coordinates, and we convert them to world coordinates in this step. Do you have thoughts on whether this should be absorbed in the grasp planning utilities instead?

Maybe worth computing both the object-centric and world-centric poses in the `Grasp` object up front?

Reply:
Well, whatever you do should be added to the Conventions section of the tutorials, I think:

https://pyrobosim.readthedocs.io/en/latest/usage/geometry_conventions.html

It makes sense (to me) to transform to `world` frame in the Grasp object. Especially since everything else (nav) is always in `world` frame, right?

Reply:
Yeah. Although another angle is if we're doing relative grasps to an object, and the object moves on us, we can still maintain that grasp. 

We'll cross that bridge when we get there, but agree that maybe world coordinates is the way to go. Will update.